Assets/Project/Scripts/Common/Components/UIs/MessageDialog.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Treevel.Common.Entities;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Treevel.Common.Components.UIs
+{
+    public class MessageDialog : MonoBehaviour
+    {
+        /// <summary>
+        /// ダイアログのタイプ
+        /// </summary>
+        public enum EDialogType {
+            Ok_Cancel, // 確定、キャンセルボタン付き
+            Ok, 　 // 確定ボタン付き
+        }
+
+        [SerializeField] private RectTransform _OkCancelTypeBtnPos;
+        [SerializeField] private RectTransform _OkTypeBtnPos;
+        [SerializeField] private MultiLanguageText _message;
+        [SerializeField] private Button _okButton;
+        [SerializeField] private MultiLanguageText _okButtonText;
+        [SerializeField] private Button _cancelButton;
+        [SerializeField] private MultiLanguageText _cancelButtonText;
+
+        /// <summary>
+        /// 外側タップする時に閉じるためのボタン
+        /// </summary>
+        [SerializeField] private Button _backgroundButton;
+
+        public void Initialize(EDialogType dialogType, ETextIndex message, ETextIndex okText, Action okCallBack)
+        {
+            // メッセージ設定
+            _message.TextIndex = message;
+
+            // OKボタンの文字
+            _okButtonText.TextIndex = okText;
+
+            // コールバック設定
+            _okButton.onClick.RemoveAllListeners();
+            _okButton.onClick.AddListener(() => {
+                // 設定したコールバックを実行
+                okCallBack?.Invoke();
+                // クリックした後閉じる
+                gameObject.SetActive(false);
+            });
+
+            // OKボタンの位置設定
+            if (dialogType == EDialogType.Ok_Cancel) {
+                _okButton.transform.SetParent(_OkCancelTypeBtnPos);
+            } else if (dialogType == EDialogType.Ok) {
+                _okButton.transform.SetParent(_OkTypeBtnPos, false);
+                _cancelButton.gameObject.SetActive(false);
+            }
+        }
```

Comment:
```suggestion
            // OKボタンの位置設定
            if (dialogType == EDialogType.Ok_Cancel) {
                _okButton.transform.SetParent(_OkCancelTypeBtnPos, false);
                _cancelButton.gameObject.SetActive(true);
            } else if (dialogType == EDialogType.Ok) {
                _okButton.transform.SetParent(_OkTypeBtnPos, false);
                _cancelButton.gameObject.SetActive(false);
            }
        }
```

- `Ok`Dialogを作った後に、`OkCancel`Dialogを作ると、キャンセルが表示されない & Okの位置が中央になる
- `Ok`Dialogを作るときに、`SetBackgroundButtonActive(false)`にすると、次にDialogを作るときもその設定が反映される

```
private void Initialize() {
    SetBackgroundButtonActive(true);
}
```
みたいなのを`public void Initialize(dialogType, message, okText, okCallback)`から呼ぶと良いかも

Reply:
>OkDialogを作った後に、OkCancelDialogを作ると、キャンセルが表示されない & Okの位置が中央になる
OkDialogを作るときに、SetBackgroundButtonActive(false)にすると、次にDialogを作るときもその設定が反映される

e0d2779 で修正